go/vt/vtgate/plan_execute.go

```diff
@@ -140,6 +141,15 @@ func (e *Executor) newExecute(
 			safeSession.RecordWarning(warning)
 		}
 
+		// set the overall query timeout if it is not already set
+		if vcursor.queryTimeout > 0 {
+			timeoutOnce.Do(func() {
+				var cancel context.CancelFunc
+				ctx, cancel = context.WithTimeout(ctx, vcursor.queryTimeout)
+				defer cancel()
```

Comment:
This is wrong: `cancel` will be both created and called instantly, because the function in `Do()` exits immediately.
I think what you wanted was to define `cancel` outside, just after the definition of `timeouOnce`, and then defer calling it if it's not nil. See code suggestion.

I'd expect a test to fail here. Can you please create a test that validates this behavior is wrong?

Reply:
I was trying something here. Definitely this is wrong.

Reply:
There's furthermore no concurrency here, so I think a simple `if cancel == nil {ctx, cancel = context.WithTimeout(ctx, vcursor.queryTimeout)}` will suffice, no need for `Once` here.